lib/vega_lite/export.ex

```diff
@@ -0,0 +1,50 @@
+defmodule VegaLite.Export do
+  @moduledoc """
+  Various export methods for a `VegaLite` specification.
+  """
+
+  require EEx
+
+  @doc """
+  Saves the specification as an HTML file.
+
+  The HTML page loads necessary JavaScript dependencies from a CDN
+  and then renders the graphic in a root element.
+  """
+  @spec save_html!(VegaLite.t(), binary()) :: :ok
+  def save_html!(vl, path) do
+    File.write!(path, to_html(vl))
+  end
+
+  defp to_html(vl) do
```

Comment:
We definitely want to make the `to_html` (and `to_pdf`, etc) functions public. That's because providing an API that allows us to access the data without reading/writing to the filesystem is going to be much more useful.

Then there are two questions:

1. Do we want to provide an API that writes to disk? Not strictly necessary but it can be handy. I also like the Python API here: VegaLite.export("foo.png") and, based on the extension, we know the format the user wants

2. Should we have this in a separate module or as part of VegaLite, especially because we already have to_json in there?

Reply:
Actually `to_html` is rather specific, because for other types we will use the npm package and always write to some file. I agree that public `to_html` can be useful.

Given we are going to have multiple output formats I didn't want to put everything in a single module. So my reasoning was that `VegaLite` should just contain the functions that directly alter the specification. JSON is specific because it's not only an output format, but actually a valid format for the specification - it makes sense to have `VegaLite.from_json`, and then `VegaLite.to_json` is there just for consistency.

Having an additional `export` function sounds good to me, but we would keep the specific functions public so they can be documented separately, right?

Reply:
> Actually to_html is rather specific, because for other types we will use the npm package and always write to some file.

I have checked and all of the other types allow reading from stdin and writing to stdout, so we should definitely expose them with a non-file based API. That's why I thought about exposing to_html, to_pdf, to_svg, etc and IO based and having the `export` function only as a high-level one for convenience.

I am still not convinced on to_json being in VegaLite while others aren't. Especially because they all have external dependencies. Except in to_json it is a Hex package but in the others it is a system executable. So I would either have them all in one place or all in export.

Reply:
Sounds good! Should we keep `export!/2` in the Export module, or have a delegate from `VegaLite`?

Reply:
I think we should keep it in the Export module. Maybe call it save! to avoid duplicating words?

Reply:
save! it is :shipit: